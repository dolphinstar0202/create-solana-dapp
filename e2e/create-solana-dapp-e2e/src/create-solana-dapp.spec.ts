import { execCapture, formatOutput } from '@solana-developers/e2e-utils'

describe('create-solana-dapp', () => {
  it('should print the help command', async () => {
    const output = execCapture(`npx --yes create-solana-dapp@e2e --help`)
    // Trim all whitespace from the output to make the snapshot more readable
    const trimmed = formatOutput(output)
    expect(trimmed).toMatchSnapshot()
  })
})
