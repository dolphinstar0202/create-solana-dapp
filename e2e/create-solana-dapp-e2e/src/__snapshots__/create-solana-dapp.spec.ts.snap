// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create-solana-dapp should print the help command 1`] = `
"â”Œ create-solana-dapp 2.0.0
Usage: create-solana-dapp [options] [name]
Arguments:
name Name of the project (default: <prompt>)
Options:
-V, --version
Output the version number
-p, --preset <preset>
Preset to use (default: <prompt>, options: next, react)
--ui <ui-library>
UI library to use (default: <prompt>)
-a, --anchor <template>
Name of the Anchor template to use (default: <prompt>, set to "none" to prevent adding Anchor)
--anchor-build
Build the anchor project (default: false)
--anchor-name <name>
Anchor project name (default: anchor)
--web-name <name>
Web project name (default: web)
-pm, --package-manager <package-manager>
Package manager to use (default: npm)
--yarn
Use yarn as the package manager (default: false)
--pnpm
Use pnpm as the package manager (default: false)
-d, --dry-run
Dry run (default: false)
-h, --help
Display help for command
Examples:
$ create-solana-dapp my-app --preset react
$ create-solana-dapp my-app --preset react --package-manager yarn
$ create-solana-dapp my-app --preset react --anchor hello-world"
`;
